livedata/src/main/java/io/getstream/chat/android/livedata/service/sync/BackgroundSyncConfigStore.kt

```diff
@@ -0,0 +1,59 @@
+package io.getstream.chat.android.livedata.service.sync
+
+import android.content.Context
+import android.content.SharedPreferences
+import androidx.security.crypto.EncryptedSharedPreferences
+import androidx.security.crypto.MasterKey
+
+interface BackgroundSyncConfigStore {
+    fun put(config: BackgroundSyncConfig)
+    fun get(): BackgroundSyncConfig
+    fun clear()
+}
+
+class EncryptedBackgroundSyncConfigStore(context: Context) : BackgroundSyncConfigStore {
+    private val prefs: SharedPreferences
+
+    init {
+        val masterKey = MasterKey.Builder(context, MASTER_KEY_ALIAS)
+            .setKeyScheme(MasterKey.KeyScheme.AES256_GCM)
+            .build()
+        prefs = EncryptedSharedPreferences.create(
+            context,
+            SYNC_CONFIG_PREFS_NAME,
+            masterKey,
+            EncryptedSharedPreferences.PrefKeyEncryptionScheme.AES256_SIV,
+            EncryptedSharedPreferences.PrefValueEncryptionScheme.AES256_GCM
+        )
+    }
+
+    override fun put(config: BackgroundSyncConfig) {
+        prefs.edit().apply {
+            putString(KEY_API_KEY, config.apiKey)
+            putString(KEY_USER_ID, config.userId)
+            putString(KEY_USER_TOKEN, config.userToken)
+        }.apply()
+    }
+
+    override fun get(): BackgroundSyncConfig = prefs.run {
+        val apiKey = getString(KEY_API_KEY, "")
+        val userId = getString(KEY_USER_ID, "")
+        val userToken = getString(KEY_USER_TOKEN, "")
+
+        if (apiKey == null || userId == null || userToken == null) {
```

Comment:
Because you are using a default parameter when ask for sharedPreferences keys, this condition will be false always. We can do something like:
```suggestion
        if (apiKey.isNullOrBlank() || userId.isNullOrBlank() || userToken.isNullOrBlank()) {
```

Reply:
This is just to satisfy lint which interprets result of getString() as String? type. 
Anyway, isNullOrBlank() is better here, so I'll use it